x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_search_bar/types.ts

```diff
@@ -15,8 +15,8 @@ export interface AlertsSearchBarProps {
   rangeFrom?: string;
   rangeTo?: string;
   query?: string;
-  onQueryChange: ({}: {
+  onQueryChange: (query: {
     dateRange: { from: string; to: string; mode?: 'absolute' | 'relative' };
-    query?: string;
+    query: string;
```

Comment:
In the implementation, we always pass a string to this function and in the observability plugin, we rely on getting a string for the query (if it is undefined, '' will be returned).